test/core/tsi/spiffe_ssl_transport_security_test.cc

```diff
@@ -0,0 +1,438 @@
+      /*expect_server_success=*/true,
+      /*expect_client_success_1_2=*/true,
+      /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Just Spiffe bundles on the client side - the server side has the root
+// configured as a certificate
+TEST_P(SpiffeSslTransportSecurityTest, ClientSideSpiffeBundle) {
+  auto* fixture = new SslTsiTestFixture(kServerKeyPath, kServerCertPath,
+                                        kClientKeyPath, kClientCertPath, "",
+                                        kClientSpiffeBundleMapPath, kCaPemPath,
+                                        /*expect_server_success=*/true,
+                                        /*expect_client_success_1_2=*/true,
+                                        /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Just SPIFFE bundles on the server side - the client side has the root
+// configured as a certificate
+TEST_P(SpiffeSslTransportSecurityTest, ServerSideSpiffeBundle) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerKeyPath, kServerCertPath, kClientKeyPath, kClientCertPath,
+      kServerSpiffeBundleMapPath, "", kCaPemPath,
+      /*expect_server_success=*/true,
+      /*expect_client_success_1_2=*/true,
+      /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Use the good client spiffebundle on the server side so we don't get a
+// matching trust domain on the server side for the client's leaf certificate
+// Client-side TLS 1.3 sees success
+TEST_P(SpiffeSslTransportSecurityTest, MTLSSpiffeServerMismatchFail) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerKeyPath, kServerCertPath, kClientKeyPath, kClientCertPath,
+      kClientSpiffeBundleMapPath, kClientSpiffeBundleMapPath, kNonSpiffeCAPath,
+      /*expect_server_success=*/false,
+      /*expect_client_success_1_2=*/false,
+      /*expected_client_success_1_3=*/true);
+  fixture->Run();
+}
+
+// Use the good server side spiffebundle on the client side so we don't get a
+// matching trust domain on the client side for the server's leaf certificate
+TEST_P(SpiffeSslTransportSecurityTest, MTLSSpiffeClientMismatchFail) {
+  auto* fixture = new SslTsiTestFixture(
+      kServerKeyPath, kServerCertPath, kClientKeyPath, kClientCertPath,
+      kServerSpiffeBundleMapPath, kServerSpiffeBundleMapPath, kNonSpiffeCAPath,
+      /*expect_server_success=*/false,
+      /*expect_client_success_1_2=*/false,
+      /*expected_client_success_1_3=*/false);
+  fixture->Run();
+}
+
+TEST_P(SpiffeSslTransportSecurityTest, NonSpiffeServerCertFail) {
+  auto* fixture = new SslTsiTestFixture(
+      kNonSpiffeKeyPath, kNonSpiffeCertPath, kClientKeyPath, kClientCertPath,
+      kServerSpiffeBundleMapPath, kClientSpiffeBundleMapPath, kNonSpiffeCAPath,
+      /*expect_server_success=*/false,
+      /*expect_client_success_1_2=*/false,
+      /*expected_client_success_1_3=*/false);
+  fixture->Run();
+}
+
+TEST_P(SpiffeSslTransportSecurityTest, NonSpiffeClientCertFail) {
```

Comment:
Suggested edit (for consistency with the other comments above):

"The server side is configured with only a SPIFFE bundle, but the client leaf certificate does not have a SPIFFE ID. When negotiating TLS 1.3, the client-side handshake succeeds because server verification of the client certificate occurs after the client-side handshake is complete."